ckanext/ogdchcommands/logic.py

```diff
@@ -160,3 +160,34 @@ def ogdch_cleanup_resources(context, data_dict):
         "count_deleted": count,
         "dryrun": dryrun,
     }
+
+
+def cleanup_package_extra(context, data_dict):
+    """
+    cleans up package_extra table for a given key
+    """
+    dryrun = data_dict.get('dryrun')
+    key = data_dict.get('key')
+    tk.check_access('package_delete', context, data_dict)
+    delete_package_extras = model.Session.query(model.PackageExtra) \
+        .filter(model.PackageExtra.key == key) \
+        .all()
+    delete_package_extra_ids = [extra.id for extra in delete_package_extras]
```

Comment:
Sorry, but I cannot find the line where you explicitly delete package_extra for a given key

Reply:
It is in these lines:

```
    sql = '''begin;
    delete from package_extra_revision
    where continuity_id in ('{delete_id_values}');
    delete from package_extra
    where id in  ('{delete_id_values}');
    commit;
    '''.format(delete_id_values="','".join(delete_package_extra_ids))
```

I am using sql to delete those since there are dependencies to the table ``package_extra_revision`` that need to get deleted first. As far as I know there is no ckan action to do this.

Does that clarify your question?